### PR TITLE
SMD and omf4

### PR DESCRIPTION
**Pull request by urbach**
This is a (preliminary) implementation of the SMD algorithm together with a fix for the OMF4 integration scheme

### CODE REVIEW

**Comment by kostrzewa** [comment]
I'm afraid this has developed some conflicts as a result of the merges that I committed today.

File: integrator.c [hunk 1]
```diff
@@ -40,13 +40,13 @@ static const double omf4_vartheta = 0.08398315262876693;
 static const double omf4_lamb = 0.6822365335719091;
 
 /* second order minimal norm integration scheme */
-void integrate_2mn(const double tau, const int S, const int halfstep);
+void integrate_2mn(const double tau, const int S, const int halfstep, const double next_tau);
```
Comment (kostrzewa) [inline integrator.c]: Have these changes to 2mn been checked for correctness in some way?

File: start.c [hunk 2]
```diff
@@ -535,45 +535,157 @@ double random_su3adj_field(const int repro, su3adj ** const momenta) {
...
+#ifdef TM_USE_MPI
+    if(g_proc_id != 0) {
+      rlxd_get(rlxd_state_backup);
+    } else if(g_proc_id == 0) {
+      rlxd_get(rlxd_state);
+    }
+    MPI_Bcast(rlxd_state, 105, MPI_INT, 0, MPI_COMM_WORLD);
+    rlxd_reset(rlxd_state);
+#endif
+    for(t0 = 0; t0 < g_nproc_t*T; t0++) {
```
Comment (kostrzewa) [inline start.c]: Is this a substantial fraction of runtime? If so, perhaps OpenMP parallelisation would be in order.

**Comment by urbach** [comment]
this is outdated

**Pull request closed without merging**